src/app/components/enemy-list/enemy-list.component.html

```diff
@@ -0,0 +1,9 @@
+<div class='container'>
+  <div class="enemyList" *ngIf="dungeon">
+    Obstacles:
+    <div class='row' *ngFor="let item of dungeon.obstacles; let i = index" [attr.data-index]="i" (click)="selectObstacle(item)">
```

Comment:
solve this without using indexes
use encapsulation principles
test for changing values

Reply:
fixed, thanks!